### ISSUE TITLE
Custom LRTR tech tree icons not showing

### ISSUE THREAD

**Issue by gotmachine** [KSPModdingLibs/KSPCommunityFixes#224]
Using the LRTR mod, forum user @Scottmm78[ is reporting some icons of the custom tech tree not showing when KSPCF is installed](https://forum.kerbalspaceprogram.com/topic/204002-18-112-kspcommunityfixes-bugfixes-and-qol-tweaks/?do=findComment&comment=4385616).

![frBjk9o](https://github.com/KSPModdingLibs/KSPCommunityFixes/assets/24925209/7fba6630-48db-4101-a02b-fbb2ff98e519)
![P8lKkyO](https://github.com/KSPModdingLibs/KSPCommunityFixes/assets/24925209/b00e54c0-f8fb-4678-980f-e6be7f5e3d2f)

User logs : [LRTR_KSP.zip](https://github.com/KSPModdingLibs/KSPCommunityFixes/files/15209663/LRTR_KSP.zip)

This seems to happen only with multiple-of-4-width/height icons, meaning those converted to DXT5.
The user is reporting the issue happening with both the texture cache enabled and disabled

I can't reproduce this locally, neither in isolation nor with the same mod set as him.

I've asked him to try a stripped down version of the mod in a minimum install : 
[LRTR_KSPCF_Test.zip](https://github.com/KSPModdingLibs/KSPCommunityFixes/files/15209622/LRTR_KSPCF_Test.zip)

Waiting for some feedback on this.

**Comment by gotmachine** [comment]
Okay, managed to reproduce this by running a minimal repro install from the steam directory, ie :
`C:\Program Files (x86)\Steam\steamapps\common\Kerbal Space Program`
In that case, the problematic textures indeed fail to load on the 
`bool ImageConversion.LoadImage(Texture2D tex, byte[] data, bool markNonReadable)`
call with the following Player.log error (example shown for a 100x100 texture with mipmaps) : 
```
d3d11: failed to create 2D texture id=1984 width=50 height=50 mips=6 dxgifmt=77 [D3D error was 80070057]
d3d11: failed to create 2D texture shader resource view id=1984 [D3D error was 80070057]
```
I actually noticed such errors popping occasionally in the past, but always failed to identify and reproduce them consistently (the fact that there is no way to find back which texture this is about from the error message doesn't help...)

My intuition tells me that I probably screwed something in the threaded file load or in how the buffers are read, will try to investigate further, but I fail to see the relationship with the location from which files/textures are loaded.

**Comment by gotmachine** [comment]
Okay, scrap all that, the issue is caused by having a lower than max global mipmap bias during loading.

**Comment by gotmachine** [comment]
So this has turned into quite the rabbit hole...

After loosing many hours trying a bunch of stuff with what unity provides, implementing a custom PNG loader, I came to the conclusion that Unity simply unable to both generate mipmaps and convert a texture to DXT5. You can have either, but not both. 

Then I asked myself, but how in hell does stock do ?
And yeah.
It doesn't.
Turns out that when I wrote the custom PNG loader, I misunderstood in which cases stock is generating mipmaps, so all that time, KSPCF was generating them when it actually shouldn't have.

**Comment by gotmachine** [comment]
Should be fixed in 1.35.1
